File: pyru/lib/pyru_examples.rb
# pyru_examples.rb

# Require the pyru_ruby_bridge gem
require 'pyru_ruby_bridge'

# Example method to demonstrate calling a Pyru method using the bridge module
def call_pyru_method
  begin
    # Call the Pyru method using the RubyBridge class
    result = PyruRubyBridge::RubyBridge.some_pyru_method("example_argument")

    # Print the result
    puts "Result from Pyru method: #{result}"
  rescue PyruRubyBridge::PyruException => e
    # Handle Pyru exceptions raised by the bridge module
    puts "Pyru exception caught: #{e.message}"
  end
end

# Main method
def main
  # Call the example method to demonstrate calling a Pyru method
  call_pyru_method
end

# Entry point of the script
if __FILE__ == $0
  main
end

File: pyru/pyru/pyru_examples.py
# pyru_examples.py

# Import the Pyru-to-Ruby bridge module
from pyru_ruby_bridge import RubyBridge, PyruException # type: ignore

# Example function to demonstrate calling a Ruby method using the bridge module
def call_ruby_method():
    try:
        # Call the Ruby method using the RubyBridge class
        result = RubyBridge.some_ruby_method("example_argument")

        # Print the result
        print("Result from Ruby method:", result)
    except PyruException as e:
        # Handle Pyru exceptions raised by the bridge module
        print("Pyru exception caught:", e)

# Main function
def main():
    # Call the example function to demonstrate calling a Ruby method
    call_ruby_method()

# Entry point of the script
if __name__ == "__main__":
    main()
